### DISCUSSION TITLE
Roadmap to remove obsolete member in BCL?

### DISCUSSION THREAD

**Discussion by MauNguyenVan** [dotnet/runtime#100680] [General] [answered]
Hi team. 
I know . net has Obsolete attribute to mark a member is deprecated. 
so. i'm curious about in roadmap these obsolete members in BCL. when these will delete or never, when a member start making obsolete, how long will it support before remove totally.i read the docs, but no information mentions about that. 
Thank you.

**Comment by stephentoub** [comment, marked as answer]
> when these will delete or never

Pretty much never, with _very_ rare exception, e.g. https://github.com/dotnet/runtime/issues/98245.

**Reply by MauNguyenVan** [reply]
What about marks a member is obsolete with error is true, is it seem the same with delete from codebase.

**Reply by stephentoub** [reply]
That happens a bit more, but still very rare. It'd be done in a situation where we don't want to break existing binaries but we also never want new binaries to use it. Most commonly these days it's used on brand new types where we're forced to expose some method but we don't actually want it to be used because it would effectively always be the wrong thing.

**Reply by MauNguyenVan** [reply]
Ok. Thank you for your support
